src/EnrollCallAndSmsController.js

```diff
@@ -274,8 +282,24 @@ function (Okta, FormController, Footer, PhoneTextBox, TextBox, CountryUtil, Form
         noButtonBar: true,
         autoSave: true,
         className: getClassName(factorType),
+
+        showWarning: function (msg) {
+          clearTimeout(warningTimeout);
+          this.clearWarnings();
+          this.add(warningTemplate, '.o-form-error-container', {options: {warning: msg}});
+        },
+        clearWarnings: function () {
+          this.$('.okta-form-infobox-warning').remove();
+        },
+
         initialize: function () {
+          this.listenTo(this.model, 'change:ableToResend', function (model, ableToResend) {
+            if (ableToResend) {
+              this.showWarning(warning);
+            }
+          });
           this.listenTo(this.model, 'error errors:clear', function () {
+            this.clearWarnings();
```

Comment:
while this probably works, we are listening to `errors` in here, not `warnings`, so I think conceptually it will be better if we don't add this line. By adding the `else` to the listener of `ableToResend` the warning should be cleared

Reply:
This is added here because warnings and errors could happen at the same time. For example, `An SMS message was recently sent. Please wait 30 seconds before trying again.` could happen at the same time with `Haven't received an SMS? To try again, click Re-send code.` I added this line to clear all the errors together